### PR TITLE
Emit file comments when only running googmodule. 

### PR DESCRIPTION
**Pull request by mprobst**
Previously, when tsickle ran only the googmodule transformer, e.g. when
both type annotation and decorator conversion is false, file level
comments would not be emitted, causing problems with some load bearing
comments. Always running at least the synthesize comments pass avoids
the issue.

### CODE REVIEW

File: src/tsickle.ts [hunk 1]
```diff
@@ -2055,6 +2055,9 @@ export function emitWithTsickle(
     ]
   };
   if (host.googmodule) {
```
[the remark this answers is not in the document]
Comment (mprobst) [reply src/tsickle.ts]: No, that doesn't work. The googmodule processor must run for all files, even if `skipTransformForSourceFileIfNeeded` is true for them, so it cannot go into the regular `after` bucket. It also shouldn't be in the `afterTs` bucket, as any additional `after` ops would then happen after it, but the `googmodule` converter really should be last.

On the bright side, I've debugged this, and was able to get rid of using `synthesizeCommentsTransformer` here (pending TAP, but I've also been able to reproduce the issue locally). So this is less messy now.

File: test_files/file_comment/comment_before_elided_import.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @fileoverview This is a comment before an import that will be elided.
```
[the remark this answers is not in the document]
Comment (mprobst) [reply test_files/file_comment/comment_before_elided_import.ts]: Clarified. The `import` is what's being elided, not the comment, which should indeed be kept (that's the point of the test).

File: test_files/file_comment.puretransform/comment_before_var.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * @fileoverview This comment must not be emitted twice.
+ * @mods {google3.java.com.google.javascript.typescript.examples.boqui.boqui}
+ * @modName {foobar}
```
Comment (LucasSloan) [inline test_files/file_comment.puretransform/comment_before_var.ts]: Is mods and modName load bearing here?  Either explain, rename test, or generate 2 tests - one for `@mods` and one for comment before a constant.

File: src/tsickle.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ import {ModulesManifest} from './modules_manifest';
 import {getEntityNameText, getIdentifierText, Rewriter, unescapeName} from './rewriter';
 import {containsInlineSourceMap, extractInlineSourceMap, parseSourceMap, removeInlineSourceMap, setInlineSourceMap, SourceMapper, SourcePosition} from './source_map_utils';
 import {createTransformerFromSourceMap} from './transformer_sourcemap';
-import {createCustomTransformers, noOpTransformer} from './transformer_util';
+import {createCustomTransformers, synthesizeCommentsTransformer} from './transformer_util';
```
Comment (LucasSloan) [inline src/tsickle.ts]: I'm not seeing a diff corresponding to this import's name being changed - unused import?

File: src/googmodule.ts [hunk 5]
```diff
@@ -166,13 +166,11 @@ export function resolveIndexShorthand(
 function importPathToGoogNamespace(
     host: GoogModuleProcessorHost, file: ts.SourceFile, tsImport: string): ts.StringLiteral {
   let modName: string;
-  let isGoogImport = false;
```
Comment (LucasSloan) [inline src/googmodule.ts]: Unused?

File: src/tsickle.ts (hunk 4 above)
Comment (mprobst) [reply src/tsickle.ts]: Good catch, done.

File: src/googmodule.ts (hunk 5 above)
Comment (mprobst) [reply src/googmodule.ts]: Yes, this was unused, and is now removed.

File: test_files/file_comment.puretransform/comment_before_var.ts (hunk 3 above)
Comment (mprobst) [reply test_files/file_comment.puretransform/comment_before_var.ts]: This is a 1:1 copy of the previously existing test (`file_comment/comment_before_var.ts`). The original test makes sure that merging a file overview comment before an exported var works. It must contain markers that it is a fileoverview comment. It does not quite matter what those are (fileoverview or mods etc), but this repros the exact problem we had in google3.

In the puretransform case, the tags in the comment are insignificant, but we generally try to keep the files identical between multiple variants (read: copies) of the same test case for different situations.

I could either change both or keep both this way I think. Given that changing them doesn't fix anything and is unrelated to this PR, I think I'd rather keep them the way they are.

**Pull request merged**